ui/app/AppLayouts/Chat/controls/community/MembersDropdown.qml

```diff
@@ -0,0 +1,237 @@
+            input.placeholder.font.pixelSize: 13
+        }
+
+        StatusBaseText {
+            id: noContactsText
+
+            Layout.fillWidth: true
+            Layout.preferredHeight: 50
+            horizontalAlignment: Text.AlignHCenter
+            verticalAlignment: Text.AlignVCenter
+
+            visible: listView.count === 0
+
+            text: qsTr("No contacts found")
+            color: Theme.palette.baseColor1
+            font.pixelSize: 12
+            elide: Text.ElideRight
+            lineHeight: 1.2
+        }
+
+        StatusListView {
+            id: listView
+
+            Layout.fillWidth: true
+            Layout.fillHeight: true
+
+            visible: count > 0
+
+            header: StatusCheckBox  {
+                width: ListView.view.width
+
+                text: qsTr("Select all")
+                font.weight: Font.Medium
+
+                checked: root.selectedKeys.length === listView.count
+
+                leftSide: false
+                size: StatusCheckBox.Size.Small
+                indicator.anchors.rightMargin: 12
+
+                MouseArea {
+                    anchors.fill: parent
+                    cursorShape: Qt.PointingHandCursor
+
+                    onClicked: {
+                        if (listView.headerItem.checked) {
+                            root.selectedKeys = []
+                            return
+                        }
+
+                        const count = root.model.rowCount()
+                        const keys = []
+
+                        for (let i = 0; i < count; i++) {
+                            const key = ModelUtils.get(root.model, i, "pubKey")
+                            keys.push(key)
+                        }
+
+                        root.selectedKeys = keys
+                    }
+                }
+            }
+
+            delegate: ContactListItemDelegate {
+                id: delegateRoot
+
+                width: ListView.view.width
+                height: d.delegateHeight
+                asset.width: 29
+                asset.height: 29
+
+                rightPadding: 0
+                leftPadding: 6
+
+                color: "transparent"
+
+                onClicked: {
+                    const index = root.selectedKeys.indexOf(model.pubKey)
+                    const selectedKeysCopy = Object.assign(
+                                               [], root.selectedKeys)
+
+                    if (index === -1)
+                        selectedKeysCopy.push(model.pubKey)
+                    else
+                        selectedKeysCopy.splice(index, 1)
+
+                    root.selectedKeys = selectedKeysCopy
+                }
+
+                components: [
+                    StatusCheckBox  {
+                        id: contactCheckbox
+
+                        size: StatusCheckBox.Size.Small
+                        checked: root.selectedKeys.indexOf(model.pubKey) > -1
+
+                        MouseArea {
```

Comment:
The main reason is that I want to control `checked` state by the given condition instead of user's click.
In theory I could set `checkable: false` and handle `onClicked` signal but the problem is somehow buggy `CheckBox` implementation:
```
                CheckBox {
                    checkable: false
                    text: checked

                    onClicked: {
                        console.log(checked)
                        checked = !checked
                    }
                }
 ```
consecutive clicks generates random output like:
```
qml: false
qml: true
qml: true
qml: false
qml: false
qml: false
qml: false
qml: false
qml: true
qml: false
```
:confused:

Reply:
You have to handle the `onToggled` signal; the moment you `click`, the internal state of `checked` may not be up to date, see: https://doc.qt.io/qt-5/qml-qtquick-controls2-abstractbutton.html#toggled-signal

Reply:
Yeah, `onToggled` allows combining interactive switching with binding to `checked`. And it also works here as well. But... in practice, while having it done this way, when "Select all" is toggled, the rest of checkboxes react with a minimal delay. It's subtle but annoying. When interactive toggling is disabled by adding additional MouseArea, toggles are done exclusively via expression bound to `checked`. As a result the changes are done synchronously, there is no delay. For that reason I would opt for leaving this implementation details as it is.

Reply:
OK, fair enough